### DISCUSSION TITLE
Talos install error - couldn't get current server API group list: - tls: internal error

### DISCUSSION THREAD

**Discussion by DavidIlie** [onedr0p/cluster-template#1407] [Q&A]
Following the pathway to install Talos, I have an issue where the cluster does seem to setup but the workers do not join, and the masters have these errors:

error refreshing pod status and the error is related to TLS (tls: internal error)
and also controller failed errors too
![image](https://github.com/onedr0p/cluster-template/assets/47594764/f5cc7b28-11cf-4898-be30-18d5a5933530)

this is what I see in my terminal while setting it up
![image](https://github.com/onedr0p/cluster-template/assets/47594764/c15aaa18-31af-4551-af57-6cc1c8252ef8)

**Comment by onedr0p** [comment]
I think this is expected due to etcd not being bootstrapped? 

https://github.com/onedr0p/cluster-template/blob/b6234fcbad3f9b3be13f94a563b6b73cbe3741f5/.taskfiles/Talos/Taskfile.yaml#L55-L63

It doesn't seem like the above was ran. At what point in the task commands did you get to and was there any errors on the client side?

**Comment by bojanraic** [comment]
In addition to @onedr0p's comments, when I tried to install Talos manually, it took some time for the master to be ready.
In both of the screenshots, uptime is only a few minutes so maybe it hasn't finished bootstrapping yet.
I went back to k3s in the meantime, but please update us on your Talos install progress via this template and I may take another stab at it when time permits. 
Good luck!

**Comment by DavidIlie** [comment]
I left it running the whole night yesterday and the same thing happened. I am also sure that I think all scripts are running and before the node first reboots/loads there are errors regarding something like a "admin" certificate

Any ideas?

**Comment by onedr0p** [comment]
Maybe give it another shot when you have a moment? Not sure what happened here to be honest could be a ton of different issues from misconfig to network issues to anything else really :/

The important bits of the config that can really go wrong if not set right are the network and disk selectors.

**Comment by DavidIlie** [comment]
Disk selectors work I believe, data is being written to the disk and network is working o I believe on all the nodes.

The error is just the "tls: internal error" every time the masters try to fetch something from their own localhost IP

**Comment by DavidIlie** [comment]
The bootstrap first begins with these errors in the console

![image](https://github.com/onedr0p/cluster-template/assets/47594764/c0e0e2fe-2933-4ed4-ba4f-c78536f67fd1)

But I believe that's when the nodes get rebooted as then it boots and continues til kubelet is healthy but the error is back:

![image](https://github.com/onedr0p/cluster-template/assets/47594764/c9fe8a27-efe2-4eb8-82da-09615b9a75e3)

And then my terminal tries to connect to the VIP and nothing happens

![image](https://github.com/onedr0p/cluster-template/assets/47594764/55f0f486-4cc7-45da-a1d3-197ed4b54c02)

**Comment by onedr0p** [comment]
I saw this in your previous config (sorry this is all I have to go on from https://github.com/onedr0p/cluster-template/issues/1398#issuecomment-2029711711)

```yaml
    networkInterfaces:
      - deviceSelector:
          hardwareAddr: ""
```

That should be the nodes mac address, are you sure this is populated? It should be in `xx:xx:xx:xx:xx:xx` format and be unique per-node.

https://github.com/onedr0p/cluster-template/blob/28ae26d3a8d1b3ea79f8552ae420faf93366dda5/config.sample.yaml#L57

**Comment by onedr0p** [comment]
I added validation on `talos_nic` [here](https://github.com/onedr0p/cluster-template/commit/35df2fb0f786065e1a8bc4dd043e0b80c74ecdff) to hopefully catch this for other people in the future.

**Comment by DavidIlie** [comment]
I already populated those, I just redacted them when I sent it here. Every single value is present

**Comment by DavidIlie** [comment]
https://github.com/onedr0p/cluster-template/assets/47594764/302633b5-723c-431d-af21-d0d041ca1c81

This is a recording of what happens

**Comment by onedr0p** [comment]
I wonder if you need to use a different type of network selector in the Talos/talhelper config or change something in the NIC settings on the VM in Proxmox?

I just hand-held someone thru the whole repo who is using bare-metal nodes and we had success after figuring out they were not setting the correct value for `talos_nic` which lead me to commit validation on that.

**Comment by DavidIlie** [comment]
Do you have an example of what I would need to do?

**Comment by onedr0p** [comment]
I am probably not the best person to ask about that as I do not use any hypervisors in my life right now 😄 

Maybe a good start is to review the [ talos proxmox docs](https://www.talos.dev/v1.6/talos-guides/install/virtualized-platforms/proxmox/) and see if everything lines up there and with the rendered config here.

Keep in mind there are a bunch of different network selectors you can use so maybe mac address is not the best with PVE? I dunno.

**Comment by bojanraic** [comment]
Hi @DavidIlie have you had a chance to give Talos another try? 
I don't use proxmox but physical x86 SBCs and I am getting similar errors to you. Every time, I end up going back to k3s. So I am hoping I could apply something similar to you if you manage to get it to install properly. 
Cheers!

**Comment by DavidIlie** [comment]
I ended up getting it running, it was because of the configuration of the network adapter, so maybe take a look if you need additional drivers for yours? I gave up with Talos because i couldn't get a specific NVIDIA driver installed, so back with debian
